modules/tensor_mechanics/src/materials/crystal_plasticity/ComputeMultipleCrystalPlasticityStress.C

```diff
@@ -229,6 +232,8 @@ ComputeMultipleCrystalPlasticityStress::updateStress(RankTwoTensor & cauchy_stre
       _models[i]->setSubstepDt(_substep_dt);
 
     // calculate F^{eigen} only when we have eigenstrain
+    _inverse_eigenstrain_deformation_grad.zero();
+    _inverse_eigenstrain_deformation_grad.addIa(1.0);
```

Comment:
Are you adding these two lines here?

Reply:
with just the `setMaterialVectorSize` method, a libmesh Assert is thrown because the elastic deformation gradient comes in as a zero tensor to the `postSolveQp` method. I suspect, and would appreciate it if @schakrabortygithub would confirm, that these lines are added to avoid that case

@schakrabortygithub would you add a regression test to restart one of the eigenstrain regression tests to make sure this change will handle those cases too please?

Reply:
@jiangwen84 Yes, those lines are due to the reason as explained by @sapitts.
I will test the restart capability and add the regression test for Eigen strain case.

Reply:
@sapitts 
I have tested the restart capability of CP-model with eigenstrain without any issue. Also added a regression test at:
modules/tensor_mechanics/test/tests/crystal_plasticity/cp_eigenstrains/thermal_eigenstrain_test_restart.i

One thing, these restart files should run after their parent files, during the regression test, to make sure that corresponding checkpoint files are available during restart.